ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -104,12 +109,34 @@ - (void)createMap
     UITapGestureRecognizer *singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleSingleTap:)];
     singleTap.delegate = self;
     [_map addGestureRecognizer:singleTap];
+    
+    // Setup offline pack notification handlers.
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(offlinePackProgressDidChange:) name:MGLOfflinePackProgressChangedNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(offlinePackDidReceiveError:) name:MGLOfflinePackErrorNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(offlinePackDidReceiveMaximumAllowedMapboxTiles:) name:MGLOfflinePackMaximumMapboxTilesReachedNotification object:nil];
 
     [self updateMap];
     [self addSubview:_map];
     [self layoutSubviews];
 }
 
+-(void)createOfflinePack:(MGLCoordinateBounds)bounds style:(NSURL*)style fromZoomLevel:(double)fromZoomLevel toZoomLevel:(double)toZoomLevel name:(NSString*)name type:(NSString*)type
+{
+    
+    id <MGLOfflineRegion> region = [[MGLTilePyramidOfflineRegion alloc] initWithStyleURL:style bounds:bounds fromZoomLevel:fromZoomLevel toZoomLevel:toZoomLevel];
+    
+    NSDictionary *userInfo = @{ @"name": name };
```

Comment:
Although the internal api allows for more flexibility, these bindings need to be a little more strict. Having it strict allows for the remove pack logic to be a little more [straight forward](https://github.com/mapbox/react-native-mapbox-gl/pull/312/files#diff-d0ace68cb3e436e36c7bb0bca305815cR216). If it was an open ended dictionary that the dev decided upon, I think it would be more difficult to know which pack to remove.